web/admin.jsp

```diff
@@ -337,13 +346,86 @@
                                     }
                                 }
+
+                                //an email update request
+                                if (request.getParameter("newEmail") != null) {
+                                    String email = request.getParameter("newEmail");
+                                    int result =  thisUser.changeEmail(email);
+                                    switch(result){
+                                        case -1:
+                                            out.print("<br><br><h3>An error has occurred.  Try again or contact the admins.</h3><br><br>");
+                                        break;
+                                        case 0:
+                                            out.print("<br><br><h3>Please provide a NEW email.</h3><br><br>");
+                                        break;
+                                        case 1:
+                                            out.print("<br><br><ul><h3>A user with this E-mail already exists.  Try a different E-mail.</h3></ul><br><br>");
+                                        break;
+                                        case 2:
+                                            out.print("<br><br><h3>Email updated!</h3><br><br>");
+                                        break;
+                                    }
+                                }
+
+                                //an username update request
+                                if (request.getParameter("newUsername") != null) {
+                                    String username = request.getParameter("newUsername");
+                                    int result =  thisUser.changeUsername(username);
+                                    switch(result){
+                                        case -1:
+                                            out.print("<br><br><h3>An error has occurred trying to change the username.  Try again or contact the admins.</h3><br><br>");
+                                        break;
+                                        case 0:
+                                            out.print("<br><br><h3>Please provide a NEW username.</h3><br><br>");
+                                        break;
+                                        case 1:
+                                            out.print("<br><br><ul><h3>A user with this E-username already exists.  Try a different Username.</h3></ul><br><br>");
+                                        break;
+                                        case 2:
+                                            out.print("<br><br><h3>Username updated!</h3><br><br>");
+                                        break;
+                                    }
+                                }
+
+                                //an fname/lname update request
+                                if (request.getParameter("newName") != null) {
```

Comment:
Only change both names at once?

Reply:
No, you submit both at once, but if blank I ignore it.  I suppose that takes away "unset".